graylog2-server/src/main/java/org/graylog2/database/entities/ScopedDbService.java

```diff
@@ -74,16 +90,23 @@ public final void ensureValidScope(E entity) {
         }
     }
 
-
     public final void ensureMutability(E entity) {
         if (!isMutable(entity)) {
             throw new IllegalArgumentException("Immutable entity cannot be modified");
         }
     }
 
+    public final void ensureDeletability(E entity) {
+        if (!isDeletable(entity)) {
+            throw new IllegalArgumentException("Non-deletable entity cannot be deleted");
+        }
+    }
+
     @Override
     public final int delete(String id) {
-        ensureMutability(get(id).orElseThrow(() -> new IllegalArgumentException("Entity not found")));
+        final E entity = get(id).orElseThrow(() -> new IllegalArgumentException("Entity not found"));
+        ensureDeletability(entity);
+        ensureMutability(entity);
```

Comment:
are we sure we want to define it this way? or do we maybe want to be able to delete non-mutable entitys?

Reply:
There is a separate `deleteImmutable` method. I assume immutable entity should only be deleted via that method.

Reply:
Ah, okay. 
Should we change the name from `deleteImmutable` to `forceDelete` or something? Because it can now
also delete deletable :-D

Reply:
I think we should rename `deleteImmutable()` to `forceDelete()` and fix the comment